tests/testsuite/file_json_error_path.rs

```diff
@@ -0,0 +1,46 @@
+#![cfg(all(feature = "json5", feature = "path_to_error"))]
+
+use serde_derive::Deserialize;
+
+use config::{Config, File, FileFormat};
+
+#[derive(Debug, Deserialize)]
+struct InnerSettings {
+    #[allow(dead_code)]
+    value: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Settings {
+    #[allow(dead_code)]
+    inner: InnerSettings,
+}
+
+#[test]
+fn test_json_foo() {
+    let c = Config::builder()
+        .add_source(File::from_str(
+            r#"
+{
+    "inner": { "value": {} }
+}
+            "#,
+            FileFormat::Json,
+        ))
+        .build()
+        .unwrap();
+
+    match c.try_deserialize_with_error_path::<Settings>() {
+        Ok(_) => panic!("expected a type error"),
+        Err(err) => {
+            let path = err.path().to_string();
+            assert_eq!(path, "inner.value");
+
+            let s = format!("{err}");
+            assert_eq!(
+                s,
+                "inner.value: invalid type: unit value, expected a string for key `inner.value`"
+            );
```

Comment:
iirc the other tests use `snapbox` for assertions

Reply:
done.